cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java

```diff
@@ -51,26 +48,24 @@ public FileMetadataCleaner(TransactionRunner transactionRunner) {
   /**
    * scans for meta data in new format which has expired the log retention.
    * @param tillTime time till which files will be deleted
-   * @param transactionTimeout transaction timeout to use for scanning entries, deleting entries.
+   * @param fileCleanupBatchSize transaction clean up batch size.
    * @return list of DeleteEntry - used to get files to delete for which metadata has already been deleted
    */
   @SuppressWarnings("ConstantConditions")
-  public List<DeletedEntry> scanAndGetFilesToDelete(final long tillTime, final int transactionTimeout) {
+  public List<DeletedEntry> scanAndGetFilesToDelete(final long tillTime, final int fileCleanupBatchSize) {
     final List<DeletedEntry> toDelete = new ArrayList<>();
 
-    // we make sure transactionTimeout is greater than TX_TIMEOUT_DISCOUNT_SECS in CDAPLogAppender check.
-    final int cutOffTransactionTime = transactionTimeout - Constants.Logging.TX_TIMEOUT_DISCOUNT_SECS;
-
     try {
       TransactionRunners.run(transactionRunner, context -> {
         StructuredTable table = context.getTable(StoreDefinition.LogFileMetaStore.LOG_FILE_META);
-        Stopwatch stopwatch = new Stopwatch().start();
 
         boolean reachedEnd = false;
         Range range = Range.all();
-        while (!reachedEnd) {
+        int count = 0;
+        // stop cleaning up if there are no files to delete or if we have reached batch size
+        while (!reachedEnd && count < fileCleanupBatchSize) {
           try (CloseableIterator<StructuredRow> iter = table.scan(range, Integer.MAX_VALUE)) {
```

Comment:
Can use fileCleanupBatchSize in the scan limit so that you don't have to count the entries here